homeassistant/components/meteoswiss/__init__.py

```diff
@@ -0,0 +1,48 @@
+"""The meteoswiss integration."""
+
+import dataclasses
+
+import meteoswiss_async
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.util.hass_dict import HassKey
+
+from .const import DOMAIN
+from .coordinator import MeteoSwissDataUpdateCoordinator
+
+
+@dataclasses.dataclass
+class MeteoSwissData:
+    """Typed shared data for this component."""
+
+    client: meteoswiss_async.MeteoSwissClient
+
+
+DOMAIN_KEY: HassKey[MeteoSwissData] = HassKey(DOMAIN)
+PLATFORMS: tuple[Platform, ...] = (Platform.WEATHER,)
+
+type MeteoSwissConfigEntry = ConfigEntry[MeteoSwissDataUpdateCoordinator]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: MeteoSwissConfigEntry) -> bool:
+    """Set up meteoswiss from a config entry."""
+
+    client = meteoswiss_async.MeteoSwissClient(session=async_get_clientsession(hass))
```

Comment:
No need for doing it here, can be moved to the `__init__` of the coordinator

Reply:
I think it would be better to have a single instance of the MeteoSwiss client and store it on the data (hence the reason for the HassKey) to store it in hass.data. This way (and overall the pattern), avoid having duplicated client instances that would duplicate authentication, caching of data, and hence having a single one would reduce boilerplate and any duplication at runtime.